src/test/op_rawleftbitshift_tests.cpp

```diff
@@ -0,0 +1,847 @@
+// Copyright (c) 2012-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <policy/policy.h>
+#include <script/interpreter.h>
+#include <script/script.h>
+
+#include <test/lcg.h>
+#include <test/util/setup_common.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <util/strencodings.h>
+
+BOOST_FIXTURE_TEST_SUITE(op_rawleftbitshift_tests, BasicTestingSetup)
+
+typedef std::vector<uint8_t> valtype;
+typedef std::vector<valtype> stacktype;
+std::array<uint32_t, 2> flagset{{0, STANDARD_SCRIPT_VERIFY_FLAGS}};
+
+static valtype FromUint64(uint64_t num) {
+    valtype num_item;
+    num_item.reserve(8);
+    for (int i = 7; i >= 0; --i) {
+        num_item.push_back((num >> (i * 8)) & 0xff);
+    }
+    return num_item;
+}
+
+static void CheckTestResultForAllFlags(const stacktype &original_stack,
+                                       const CScript &script,
+                                       const stacktype &expected) {
+    BaseSignatureChecker sigchecker;
+    for (uint32_t flags : flagset) {
+        ScriptError err = ScriptError::OK;
+        stacktype stack{original_stack};
+        bool r = EvalScript(stack, script, flags, sigchecker, &err);
+        BOOST_CHECK(r);
```

Comment:
You should use BOOST_CHECK_MESSAGE here to ensure we can find failing tests in the future.